tower-http/src/services/fs/serve_dir/tests.rs

```diff
@@ -252,6 +252,50 @@ async fn missing_precompressed_variant_fallbacks_to_uncompressed_for_head_reques
     assert!(res.into_body().frame().await.is_none());
 }
 
+#[tokio::test]
+async fn precompressed_without_extension() {
+    let svc = ServeDir::new("../test-files").precompressed_gzip();
+
+    let request = Request::builder()
+        .uri("/extensionless_precompressed")
+        .header("Accept-Encoding", "gzip")
+        .body(Body::empty())
+        .unwrap();
+    let res = svc.oneshot(request).await.unwrap();
+
+    assert_eq!(res.status(), StatusCode::OK);
+
+    assert_eq!(res.headers()["content-type"], "application/octet-stream");
+    assert_eq!(res.headers()["content-encoding"], "gzip");
+
+    let body = res.into_body().collect().await.unwrap().to_bytes();
+    let mut decoder = GzDecoder::new(&body[..]);
+    let mut decompressed = String::new();
+    decoder.read_to_string(&mut decompressed).unwrap();
+    assert!(decompressed.starts_with("Content."));
```

Comment:
Why assert it _starts with_ that? This seems like it should be an `assert_eq!` - maybe the string literal needs an extra `\n` in that case, or we could call `decompressed.trim()`.

Reply:
I agree that this is a weird way to do this. I did it for consistency with the existing precompression tests, which do the same.

Reply:
Weird. No need to be consistent with that. I would welcome another PR to change the existing tests too, if you feel like it.

Reply:
I changed it to check the entire response body.

I'm afraid it will break on Windows if git is configured to change line endings to CRLF. This conversion will also only affect the uncompressed test files, which is annoying.

Reply:
I realized that I can just remove the trailing newline from the test file, so I did that. That makes it not a valid Unix text file, but there is no reason it should be.